crates/settings/src/gui.rs

```diff
@@ -337,6 +337,78 @@ impl Gui {
+                                    {
+                                        if binding == field_binding {
+                                            return match optional_vec(field, "presets") {
+                                                Some(presets) => {
+                                                    let preset_vec = presets.iter().enumerate()
+                                                        .map(|(preset_index, preset_yaml)| {
+                                                            let name = optional_string(preset_yaml, "name");
+                                                            let value = require_string(
+                                                                preset_yaml,
+                                                                Some("value"),
+                                                                Some(format!(
+                                                                    "preset value must be a string for preset index: {preset_index} for field index: {field_index} in gui deployment: {deployment_key}",
+                                                                ))
+                                                            )?;
+
+                                                            Ok(GuiPreset {
+                                                                id: preset_index.to_string(),
+                                                                name,
+                                                                value,
+                                                            })
+                                                        })
+                                                        .collect::<Result<Vec<_>, YamlError>>()?;
+                                                    Ok(Some(preset_vec))
+                                                }
+                                                None => Ok(None),
+                                            };
+                                        }
+                                    } else {
+                                        return Err(YamlError::ParseError(format!(
+                                            "binding string missing for field index: {field_index} in gui deployment: {deployment_key}",
+                                        )));
+                                    }
+                                }
+                            }
+                        } else {
+                            return Err(YamlError::ParseError(format!(
+                                "fields list missing in gui deployment: {deployment_key}"
+                            )));
+                        }
+                    }
+                } else {
+                    return Err(YamlError::ParseError(
+                        "deployments field must be a map in gui".to_string(),
```

Comment:
the idea that we're looking for a map with a key foo, and if we find foo but it's a list, seems like a reusable thing and the error could have the key foo replaced in it expected {foo} to be a map, found a list instead